website/static/js/osfHelpers.js

```diff
@@ -844,6 +844,18 @@ var trackClick = function(category, action, label){
     return true;
 };
 
+
+// Call a function when scrolled to the bottom of the element
+/// Useful for triggering an event at the bottom of a window, like infinite scroll
+function onScrollToBottom(element, callback) {
+    $(element).scroll(function() {
+        var $this = $(this);
+        if ($this.scrollTop() + $this.innerHeight() >= $this[0].scrollHeight) {
```

Comment:
This may be a little excessive, but since you had so much trouble with this, it might be a good idea to pull this boolean out into its own function and then unit test that function. If you could throw all the things that you ran into along the way that didn't work into the test, that would save future folks who try to make this fancier from messing everything up.

Alternately, you could try unit testing this function, but you'd have to fake the element properly, which might not be so bad.